policy/policy.go

```diff
@@ -38,110 +40,84 @@ type List struct {
 	Policies      map[string]TLSPolicy `json:"policies"`
 }
 
-// Get retrieves the TLSPolicy for a domain, and resolves
+// get retrieves the TLSPolicy for a domain, and resolves
 // aliases if they exist.
-func (t List) Get(domain string) (TLSPolicy, error) {
-	policy, ok := t.Policies[domain]
+func (l list) get(domain string) (TLSPolicy, error) {
+	policy, ok := l.Policies[domain]
 	if !ok {
-		return TLSPolicy{}, fmt.Errorf("Policy for %d doesn't exist")
+		return TLSPolicy{}, fmt.Errorf("Policy for %s doesn't exist", domain)
 	}
 	if len(policy.PolicyAlias) > 0 {
-		policy, ok = t.PolicyAliases[policy.PolicyAlias]
+		policy, ok = l.PolicyAliases[policy.PolicyAlias]
 		if !ok {
-			return TLSPolicy{}, fmt.Errorf("Policy alias for %d doesn't exist")
+			return TLSPolicy{}, fmt.Errorf("Policy alias for %s doesn't exist", domain)
 		}
 	}
 	return policy, nil
 }
 
-type listFetcher func(string) (List, error)
-
-// UpdatedList wraps a List that is updated from a remote
+// UpdatedList wraps a list that is updated from a remote
 // policyURL every hour. Safe for concurrent calls to `Get`.
 type UpdatedList struct {
-	messages        chan policyRequest
-	updateFrequency time.Duration
-	fetch           listFetcher
-	policyURL       string
+	mu sync.RWMutex
+	list
+}
+
+// Get safely reads from the underlying policy list and returns a TLSPolicy for a domain
+func (l UpdatedList) Get(domain string) (TLSPolicy, error) {
+	l.mu.RLock()
+	defer l.mu.RUnlock()
+	return l.get(domain)
```

Comment:
also just wanted to point out-- this is pretty cool! I had to second-take for a second. :)